src/nvim/api/buffer.c

```diff
@@ -491,6 +491,37 @@ void nvim_buf_set_lines(uint64_t channel_id,
   try_end(err);
 }
 
+/// Return the byte offset for a line.
+//
+/// This includes the end-of-line character, depending on the 'fileformat'
+/// option for the current buffer.  The first line returns 0.
+/// UTF-8 encoding is used, 'fileencoding' is ignored.
+/// Sending in the index just below the last line gives
+/// the total byte count for the entire file. A final newline
+/// will be included if it would be written, see |eol|.
+///
+/// Unlike |line2byte|, throws error for out-of-bounds indexing.
+///
+/// @param buffer     Buffer handle
+/// @param index      Line index
+/// @param[out] err   Error details, if any
+/// @return Integer   Byte offset
+Integer nvim_buf_get_offset_for_line(Buffer buffer, Integer index, Error *err)
+  FUNC_API_SINCE(5)
+{
+  buf_T *buf = find_buffer_by_handle(buffer, err);
+  if (!buf) {
+    return 0;
+  }
+
+  if (index < 0 || index > buf->b_ml.ml_line_count) {
+    api_set_error(err, kErrorTypeValidation, "Index out of bounds");
```

Comment:
Should we also return a sentinel for unloaded buffer? https://github.com/neovim/neovim/blob/ff5ab82c55be76338f8bdfbff8047796c7b33e3c/src/nvim/api/buffer.c#L247

Reply:
Done so.